glide-core/redis-rs/redis/src/pipeline.rs

```diff
@@ -323,7 +322,7 @@ macro_rules! implement_pipeline_commands {
                     0 => panic!("No command on stack"),
                     x => x - 1,
                 };
-                &mut self.commands[idx]
+                Arc::make_mut(&mut self.commands[idx])
```

Comment:
make_mut returns a `clone` of the Arc inner value if the reference count > 1. Looking at the code it seems like this function is called when `arg()` is called, e.g. with the pipeline builder. So if from any reason the rc > 1 it will break the builder. Since now you can pass to the pipeline constructor an Arc<cmd>, it can be referenced from two placed. Can you write a test for that - see where it is possible that we'll get into this issue, and how it can be restricted (in general I suggest using get_mut instead, and raising an error if it returns None - but it might require deeper changes)

Reply:
check my assumption " Since now you can pass to the pipeline constructor an Arc,' maybe it isn't possible with the current code, and then maybe using get_mute with 'expect' is legit as long as we have a test that will fail if someone will try to change the code to allow this behavior